reference/forms/types/dateinterval.rst

```diff
@@ -0,0 +1,325 @@
+|                      | - `with_minutes`_                                                                 |
+|                      | - `with_months`_                                                                  |
+|                      | - `with_seconds`_                                                                 |
+|                      | - `with_weeks`_                                                                   |
+|                      | - `with_years`_                                                                   |
+|                      | - `years`_                                                                        |
++----------------------+-----------------------------------------------------------------------------------+
+| Inherited            | - `data`_                                                                         |
+| options              | - `disabled`_                                                                     |
+|                      | - `inherit_data`_                                                                 |
+|                      | - `invalid_message`_                                                              |
+|                      | - `invalid_message_parameters`_                                                   |
+|                      | - `mapped`_                                                                       |
++----------------------+-----------------------------------------------------------------------------------+
+| Parent type          | :doc:`form </reference/forms/types/form>`                                         |
++----------------------+-----------------------------------------------------------------------------------+
+| Class                | :class:`Symfony\\Component\\Form\\Extension\\Core\\Type\\DateIntervalType`        |
++----------------------+-----------------------------------------------------------------------------------+
+
+Basic Usage
+-----------
+
+This field type is highly configurable, but easy to use. The most important
+options are ``input`` and ``widget``.
+
+Suppose that you have a ``remindEvery`` field whose underlying interval is a
+``DateInterval`` object. The following configures the ``dateinterval`` type
+for that field as **three different choice fields**:
+
+.. code-block:: php
+
+    $builder->add('remindEvery', DateIntervalType::class, array(
+        'widget' => 'choice',
+    ));
+
+The ``input`` option *must* be changed to match the type of the underlying
+interval data. For example, if the ``remindEvery`` field's data were a
+ISO 8601 duration string you'd need to set ``input`` to ``string``:
+
+.. code-block:: php
+
+    $builder->add('remindEvery', DateIntervalType::class, array(
+        'input'  => 'string',
+        'widget' => 'choice',
+    ));
+
+The field also supports an ``array`` as valid ``input`` option values.
```

Comment:
How about we replace this whole section with a bigger, example:

```rst
You can configure *a lot* of different options, including exactly *which* range
options to show (e.g. don't show "months", but *do* show "days")::

$builder->add('remindEvery', DateIntervalType::class, array(
    'widget'      => 'integer', // render a text field for each part
    // 'input'    => 'string',  // if you want the field to return a ISO 8601 string back to you

    // customize which text boxes are shown
    'with_years'  => false,
    'with_months' => false,
    'with_days'   => true,
    'with_hours'  => true,
));
```

The `with_*` stuff seems a bit weird to use :). Did you consider adding something like a `components` option instead? e.g.

```php
'components' => ['days', 'hours', 'minutes']
```